usr/share/rear/conf/default.conf

```diff
@@ -334,6 +334,28 @@ CDROM_SIZE=20
 # does not automatically recreate the rescue/recovery system.
 CHECK_CONFIG_FILES=( '/etc/drbd/' '/etc/drbd.conf' '/etc/lvm/lvm.conf' '/etc/multipath.conf' '/etc/rear/' '/etc/udev/udev.conf' )
 
+# FILES_TO_PATCH_PATTERNS is a space-separated list of shell glob patterns.
+# Files that match are eligible for a final migration of UUIDs and other
+# identifiers after recovery (if the layout recreation process has led
+# to a change of an UUID or a device name and a corresponding change needs
+# to be performed on restored configuration files ).
+# See finalize/GNU/Linux/280_migrate_uuid_tags.sh
+# The funny [] around the first letter make sure that shopt -s nullglob removes this file from the list if it does not exist
+# the files without a [] are mandatory, like fstab FIXME: but below there is [e]tc/fstab not etc/fstab - why?
+
+FILES_TO_PATCH_PATTERNS="[b]oot/{grub.conf,menu.lst,device.map} [e]tc/grub.* \
+                         [b]oot/grub/{grub.conf,grub.cfg,menu.lst,device.map} \
+                         [b]oot/grub2/{grub.conf,grub.cfg,menu.lst,device.map} \
+                         [e]tc/sysconfig/grub [e]tc/sysconfig/bootloader \
+                         [e]tc/lilo.conf [e]tc/elilo.conf \
+                         [e]tc/yaboot.conf \
+                         [e]tc/mtab [e]tc/fstab \
```

Comment:
I think `[e]tc/crypttab` should be added here because it contains UUIDs.
E.g. on my homeoffice laptop I have
```
# cat /etc/crypttab 
cr_ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part2  UUID=54fc77c5-8ec2-457f-b558-9deda3b843b2
cr_ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part3  UUID=a6dba0d8-5be8-4970-b1e7-a272ae0cafdd

# lsblk -ipo NAME,KNAME,FSTYPE,MOUNTPOINT,UUID
NAME                                                      KNAME     FSTYPE      MOUNTPOINT       UUID
/dev/sda                                                  /dev/sda                               
|-/dev/sda1                                               /dev/sda1                              
|-/dev/sda2                                               /dev/sda2 crypto_LUKS                  54fc77c5-8ec2-457f-b558-9deda3b843b2
| `-/dev/mapper/cr_ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part2 /dev/dm-1 swap        [SWAP]           6d8f8998-dd20-412a-bcc2-618eed858662
|-/dev/sda3                                               /dev/sda3 crypto_LUKS                  a6dba0d8-5be8-4970-b1e7-a272ae0cafdd
| `-/dev/mapper/cr_ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part3 /dev/dm-0 ext4        /                f05af948-6075-40a3-9191-354b0a0a9afc
|-/dev/sda4                                               /dev/sda4 ext4        /nfs             4c4a923d-1562-4254-a1fa-4e761278c02f
|-/dev/sda5                                               /dev/sda5 ext4        /var/lib/libvirt 4a42395e-4f9d-4056-9948-6d5d9d92d990
|-/dev/sda6                                               /dev/sda6 ext2                         259dac9c-f2fd-4181-a351-83603398e465
|-/dev/sda7                                               /dev/sda7 crypto_LUKS                  1b4198c9-d9b0-4c57-b9a3-3433e391e706
`-/dev/sda8                                               /dev/sda8 crypto_LUKS                  3e874a28-7415-4f8c-9757-b3f28a96c4d2
```

Reply:
I suspect there may be more files that need UUID migration indeed.

See also the remarks at the end of the commit message of f3072a38f180d540435a295acda83b1956c81c42 .

Reply:
As far as I see we have currently two different kind of disk IDs migration code:

Migrating UUIDs (i.e. things in /dev/disk/by-uuid/)
versus migrating IDs (i.e. things in /dev/disk/by-id/).

We do not have code that deals with PARTUUID (i.e. things in /dev/disk/by-partuuid/)
because 'partuuid' only appears in a comment in lib/layout-functions.sh.

There is a typo in the remarks at the end of the commit message of
https://github.com/rear/rear/commit/f3072a38f180d540435a295acda83b1956c81c42

Actually it is finalize/default/880_check_for_mount_by_id.sh
that prints the "WARNING ! You are mounting some devices by ID. ..."

The whole /dev/disk/by-id migration code logic is mostly unknown to me.
I only did some generic cleanup things but I never touched its basic logic.

I don't know if the files that need ID migration
are the same as the files that need UUID migration.

I guess it should not hurt to have a common superset of files
that need ID migration and/or UUID migration
specified via FILES_TO_PATCH_PATTERNS
and use that for both ID migration and UUID migration.

I hope the ID migration code and the UUID migration code
behaves sufficiently fail-safe when it is run with arbitrary files.

In particular it should not try to change files (i.e. run 'sed' on files)
that do not contain the ID or UUID which should be changed
(as far as I see this is missing in 280_migrate_uuid_tags.sh
 and in 260_rename_diskbyid.sh)
and it should avoid running 'sed' when old and new ID or UUID are same, cf.
https://github.com/rear/rear/commit/eb2d4a31fd89715108bb9462b9a2e9edc1e7769
(I think in 280_migrate_uuid_tags.sh FS_UUID_MAP has only entries
 when the UUID changed i.e. when old and new UUID are not same).

I think unifying ID migration and UUID migration
to let both use one same FILES_TO_PATCH_PATTERNS
is not needed in this pull request here.
Better enhance it step by step with separated pull requests.

By the way:
Usually when I try to get a better understanding of old code I call e.g.
```
# git log --follow -p usr/share/rear/finalize/GNU/Linux/260_rename_diskbyid.sh

# git log --follow -p usr/share/rear/finalize/default/880_check_for_mount_by_id.sh
```
and skim through its output.
Here the output of the first indicates
that the /dev/disk/by-id migration originated at
https://github.com/rear/rear/issues/22

FYI
what I have on my homeoffice laptop in /dev/disk/by-*
```
# lsblk -ipo NAME,KNAME,UUID,PTUUID,PARTUUID,WWN
NAME         KNAME     UUID                                 PTUUID                               PARTUUID                             WWN
/dev/sda     /dev/sda                                       9be0015e-cf90-4c6b-80ac-c4ea89832553                                      0x5000039462b83c55
|-/dev/sda1  /dev/sda1                                      9be0015e-cf90-4c6b-80ac-c4ea89832553 4a795619-730c-4e74-854b-59ea03e4f1ab 0x5000039462b83c55
|-/dev/sda2  /dev/sda2 54fc77c5-8ec2-457f-b558-9deda3b843b2 9be0015e-cf90-4c6b-80ac-c4ea89832553 ab0c1a1e-729a-4f9d-9961-9a090fc850f4 0x5000039462b83c55
| `-/dev/mapper/cr_ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part2
|            /dev/dm-1 6d8f8998-dd20-412a-bcc2-618eed858662                                                                           
|-/dev/sda3  /dev/sda3 a6dba0d8-5be8-4970-b1e7-a272ae0cafdd 9be0015e-cf90-4c6b-80ac-c4ea89832553 27533ab0-2616-4e49-89c3-8d7158636d30 0x5000039462b83c55
| `-/dev/mapper/cr_ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part3
|            /dev/dm-0 f05af948-6075-40a3-9191-354b0a0a9afc                                                                           
|-/dev/sda4  /dev/sda4 4c4a923d-1562-4254-a1fa-4e761278c02f 9be0015e-cf90-4c6b-80ac-c4ea89832553 047a5693-3b2f-4586-9d9f-5b4b25680f81 0x5000039462b83c55
|-/dev/sda5  /dev/sda5 4a42395e-4f9d-4056-9948-6d5d9d92d990 9be0015e-cf90-4c6b-80ac-c4ea89832553 a42060c3-33d5-4986-8a9f-41753ab83aa1 0x5000039462b83c55
|-/dev/sda6  /dev/sda6 259dac9c-f2fd-4181-a351-83603398e465 9be0015e-cf90-4c6b-80ac-c4ea89832553 630abca0-e067-44a0-bc97-8d8fac21b067 0x5000039462b83c55
|-/dev/sda7  /dev/sda7 1b4198c9-d9b0-4c57-b9a3-3433e391e706 9be0015e-cf90-4c6b-80ac-c4ea89832553 1ffae0c8-1212-428c-bf9e-4c3e635763bb 0x5000039462b83c55
`-/dev/sda8  /dev/sda8 3e874a28-7415-4f8c-9757-b3f28a96c4d2 9be0015e-cf90-4c6b-80ac-c4ea89832553 3af73460-4ed7-4000-b471-7c1eb35b6bc7 0x5000039462b83c55
/dev/sr0     /dev/sr0                                                                                                                 0x5001480000000000

# find /dev/disk/by-* -type l -ls | cut -b73-
/dev/disk/by-id/dm-uuid-CRYPT-LUKS1-a6dba0d85be84970b1e7a272ae0cafdd-cr_ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part3 -> ../../dm-0
/dev/disk/by-id/dm-name-cr_ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part3 -> ../../dm-0
/dev/disk/by-id/dm-uuid-CRYPT-LUKS1-54fc77c58ec2457fb5589deda3b843b2-cr_ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part2 -> ../../dm-1
/dev/disk/by-id/dm-name-cr_ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part2 -> ../../dm-1
/dev/disk/by-id/scsi-35000039462b83c55-part1 -> ../../sda1
/dev/disk/by-id/ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part1 -> ../../sda1
/dev/disk/by-id/scsi-1ATA_TOSHIBA_MQ01ABF050_Y2PLP02CT-part1 -> ../../sda1
/dev/disk/by-id/wwn-0x5000039462b83c55-part1 -> ../../sda1
/dev/disk/by-id/scsi-0ATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part1 -> ../../sda1
/dev/disk/by-id/scsi-SATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part1 -> ../../sda1
/dev/disk/by-id/scsi-SATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part5 -> ../../sda5
/dev/disk/by-id/ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part5 -> ../../sda5
/dev/disk/by-id/scsi-0ATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part5 -> ../../sda5
/dev/disk/by-id/scsi-1ATA_TOSHIBA_MQ01ABF050_Y2PLP02CT-part5 -> ../../sda5
/dev/disk/by-id/wwn-0x5000039462b83c55-part5 -> ../../sda5
/dev/disk/by-id/scsi-35000039462b83c55-part5 -> ../../sda5
/dev/disk/by-id/ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part4 -> ../../sda4
/dev/disk/by-id/scsi-1ATA_TOSHIBA_MQ01ABF050_Y2PLP02CT-part4 -> ../../sda4
/dev/disk/by-id/scsi-0ATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part4 -> ../../sda4
/dev/disk/by-id/scsi-35000039462b83c55-part4 -> ../../sda4
/dev/disk/by-id/wwn-0x5000039462b83c55-part4 -> ../../sda4
/dev/disk/by-id/scsi-SATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part4 -> ../../sda4
/dev/disk/by-id/scsi-0ATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part6 -> ../../sda6
/dev/disk/by-id/wwn-0x5000039462b83c55-part6 -> ../../sda6
/dev/disk/by-id/scsi-1ATA_TOSHIBA_MQ01ABF050_Y2PLP02CT-part6 -> ../../sda6
/dev/disk/by-id/scsi-SATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part6 -> ../../sda6
/dev/disk/by-id/scsi-35000039462b83c55-part6 -> ../../sda6
/dev/disk/by-id/ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part6 -> ../../sda6
/dev/disk/by-id/scsi-0ATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part8 -> ../../sda8
/dev/disk/by-id/scsi-35000039462b83c55-part8 -> ../../sda8
/dev/disk/by-id/ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part8 -> ../../sda8
/dev/disk/by-id/scsi-1ATA_TOSHIBA_MQ01ABF050_Y2PLP02CT-part8 -> ../../sda8
/dev/disk/by-id/scsi-SATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part8 -> ../../sda8
/dev/disk/by-id/wwn-0x5000039462b83c55-part8 -> ../../sda8
/dev/disk/by-id/scsi-SATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part7 -> ../../sda7
/dev/disk/by-id/scsi-0ATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part7 -> ../../sda7
/dev/disk/by-id/wwn-0x5000039462b83c55-part7 -> ../../sda7
/dev/disk/by-id/scsi-1ATA_TOSHIBA_MQ01ABF050_Y2PLP02CT-part7 -> ../../sda7
/dev/disk/by-id/ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part7 -> ../../sda7
/dev/disk/by-id/scsi-35000039462b83c55-part7 -> ../../sda7
/dev/disk/by-id/wwn-0x5000039462b83c55-part3 -> ../../sda3
/dev/disk/by-id/scsi-1ATA_TOSHIBA_MQ01ABF050_Y2PLP02CT-part3 -> ../../sda3
/dev/disk/by-id/scsi-35000039462b83c55-part3 -> ../../sda3
/dev/disk/by-id/ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part3 -> ../../sda3
/dev/disk/by-id/scsi-0ATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part3 -> ../../sda3
/dev/disk/by-id/scsi-SATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part3 -> ../../sda3
/dev/disk/by-id/scsi-1ATA_TOSHIBA_MQ01ABF050_Y2PLP02CT-part2 -> ../../sda2
/dev/disk/by-id/scsi-35000039462b83c55-part2 -> ../../sda2
/dev/disk/by-id/scsi-SATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part2 -> ../../sda2
/dev/disk/by-id/scsi-0ATA_TOSHIBA_MQ01ABF0_Y2PLP02CT-part2 -> ../../sda2
/dev/disk/by-id/ata-TOSHIBA_MQ01ABF050_Y2PLP02CT-part2 -> ../../sda2
/dev/disk/by-id/wwn-0x5000039462b83c55-part2 -> ../../sda2
/dev/disk/by-id/wwn-0x5001480000000000 -> ../../sr0
/dev/disk/by-id/ata-hp_DVDRAM_GU70N_M49CBQG2454 -> ../../sr0
/dev/disk/by-id/scsi-1ATA_TOSHIBA_MQ01ABF050_Y2PLP02CT -> ../../sda
/dev/disk/by-id/ata-TOSHIBA_MQ01ABF050_Y2PLP02CT -> ../../sda
/dev/disk/by-id/scsi-35000039462b83c55 -> ../../sda
/dev/disk/by-id/scsi-0ATA_TOSHIBA_MQ01ABF0_Y2PLP02CT -> ../../sda
/dev/disk/by-id/wwn-0x5000039462b83c55 -> ../../sda
/dev/disk/by-id/scsi-SATA_TOSHIBA_MQ01ABF0_Y2PLP02CT -> ../../sda
/dev/disk/by-partlabel/other -> ../../sda6
/dev/disk/by-partlabel/playground2 -> ../../sda8
/dev/disk/by-partlabel/playground -> ../../sda7
/dev/disk/by-partuuid/4a795619-730c-4e74-854b-59ea03e4f1ab -> ../../sda1
/dev/disk/by-partuuid/a42060c3-33d5-4986-8a9f-41753ab83aa1 -> ../../sda5
/dev/disk/by-partuuid/047a5693-3b2f-4586-9d9f-5b4b25680f81 -> ../../sda4
/dev/disk/by-partuuid/630abca0-e067-44a0-bc97-8d8fac21b067 -> ../../sda6
/dev/disk/by-partuuid/3af73460-4ed7-4000-b471-7c1eb35b6bc7 -> ../../sda8
/dev/disk/by-partuuid/1ffae0c8-1212-428c-bf9e-4c3e635763bb -> ../../sda7
/dev/disk/by-partuuid/27533ab0-2616-4e49-89c3-8d7158636d30 -> ../../sda3
/dev/disk/by-partuuid/ab0c1a1e-729a-4f9d-9961-9a090fc850f4 -> ../../sda2
/dev/disk/by-path/pci-0000:00:1f.2-ata-5.0-part1 -> ../../sda1
/dev/disk/by-path/pci-0000:00:1f.2-ata-5-part1 -> ../../sda1
/dev/disk/by-path/pci-0000:00:1f.2-ata-5.0-part5 -> ../../sda5
/dev/disk/by-path/pci-0000:00:1f.2-ata-5-part5 -> ../../sda5
/dev/disk/by-path/pci-0000:00:1f.2-ata-5.0-part4 -> ../../sda4
/dev/disk/by-path/pci-0000:00:1f.2-ata-5-part4 -> ../../sda4
/dev/disk/by-path/pci-0000:00:1f.2-ata-5-part6 -> ../../sda6
/dev/disk/by-path/pci-0000:00:1f.2-ata-5.0-part6 -> ../../sda6
/dev/disk/by-path/pci-0000:00:1f.2-ata-5-part8 -> ../../sda8
/dev/disk/by-path/pci-0000:00:1f.2-ata-5.0-part8 -> ../../sda8
/dev/disk/by-path/pci-0000:00:1f.2-ata-5.0-part7 -> ../../sda7
/dev/disk/by-path/pci-0000:00:1f.2-ata-5-part7 -> ../../sda7
/dev/disk/by-path/pci-0000:00:1f.2-ata-5.0-part3 -> ../../sda3
/dev/disk/by-path/pci-0000:00:1f.2-ata-5-part3 -> ../../sda3
/dev/disk/by-path/pci-0000:00:1f.2-ata-5-part2 -> ../../sda2
/dev/disk/by-path/pci-0000:00:1f.2-ata-5.0-part2 -> ../../sda2
/dev/disk/by-path/pci-0000:00:1f.2-ata-6.0 -> ../../sr0
/dev/disk/by-path/pci-0000:00:1f.2-ata-6 -> ../../sr0
/dev/disk/by-path/pci-0000:00:1f.2-ata-5 -> ../../sda
/dev/disk/by-path/pci-0000:00:1f.2-ata-5.0 -> ../../sda
/dev/disk/by-uuid/f05af948-6075-40a3-9191-354b0a0a9afc -> ../../dm-0
/dev/disk/by-uuid/6d8f8998-dd20-412a-bcc2-618eed858662 -> ../../dm-1
/dev/disk/by-uuid/4a42395e-4f9d-4056-9948-6d5d9d92d990 -> ../../sda5
/dev/disk/by-uuid/4c4a923d-1562-4254-a1fa-4e761278c02f -> ../../sda4
/dev/disk/by-uuid/259dac9c-f2fd-4181-a351-83603398e465 -> ../../sda6
/dev/disk/by-uuid/3e874a28-7415-4f8c-9757-b3f28a96c4d2 -> ../../sda8
/dev/disk/by-uuid/1b4198c9-d9b0-4c57-b9a3-3433e391e706 -> ../../sda7
/dev/disk/by-uuid/a6dba0d8-5be8-4970-b1e7-a272ae0cafdd -> ../../sda3
/dev/disk/by-uuid/54fc77c5-8ec2-457f-b558-9deda3b843b2 -> ../../sda2
```

Reply:
> I think unifying ID migration and UUID migration
> to let both use one same FILES_TO_PATCH_PATTERNS
> is not needed in this pull request here.
> Better enhance it step by step with separated pull requests.

I certainly do not intend to address the other case(s) of ID migration here. I will address your review comments shortly and finish this PR. (Question is, whether I should add `crypttab` to the migrated files in this PR, as it is also only distantly related to the original topic.)

Reply:
In default.conf I added `[e]tc/crypttab` to FILES_TO_PATCH_PATTERNS via
https://github.com/rear/rear/commit/58ddf3e094456df3c41f0b2089ee962d34836da6